robot.py

```diff
@@ -292,9 +258,15 @@ def testPeriodic(self):
                     ctre.ControlMode.Position, module.steer_enc_offset
                 )
 
-        if self.joystick.getRawButton(11):
-            for module in self.chassis.modules:
-                module.drive_motor.set(ctre.ControlMode.PercentOutput, 0.3)
+    def closest_field_angle(self, robot_heading):
+        lowest_distance = 2 * math.pi
+        lowest_label = None
+        for label, angle in self.field_angles.items():
+            diff = constrain_angle(constrain_angle(robot_heading) - angle)
+            if abs(diff) < lowest_distance:
+                lowest_distance = abs(diff)
+                lowest_label = label
+        return lowest_label
```

Comment:
Instead of trying to rewrite the `min` function would it be easier to do something like this:
```suggestion
        label, angle = min(
            self.field_angles.items(),
            key=lambda a: abs(constrain_angle(constrain_angle(robot_heading) - a[1])),
        )
        return label
```